File: src/main/java/no/nav/bidrag/beregn/forskudd/core/ForskuddCore.java
package no.nav.bidrag.beregn.forskudd.core;

import no.nav.bidrag.beregn.forskudd.core.dto.BeregnForskuddGrunnlagCore;
import no.nav.bidrag.beregn.forskudd.core.dto.BeregnetForskuddResultatCore;
import no.nav.bidrag.beregn.forskudd.core.periode.ForskuddPeriode;

public interface ForskuddCore {

  BeregnetForskuddResultatCore beregnForskudd(BeregnForskuddGrunnlagCore grunnlag);

  static ForskuddCore getInstance() {
    return new ForskuddCoreImpl(ForskuddPeriode.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/forskudd/core/ForskuddCore.kt
package no.nav.bidrag.beregn.forskudd.core

import no.nav.bidrag.beregn.forskudd.core.dto.BeregnForskuddGrunnlagCore
import no.nav.bidrag.beregn.forskudd.core.dto.BeregnetForskuddResultatCore
import no.nav.bidrag.beregn.forskudd.core.periode.ForskuddPeriode

fun interface ForskuddCore {
    fun beregnForskudd(grunnlag: BeregnForskuddGrunnlagCore): BeregnetForskuddResultatCore

    companion object {
        fun getInstance(): ForskuddCore {
            return ForskuddCoreImpl(ForskuddPeriode.getInstance())
        }
    }
}
